src/app/services/cart.service.ts

```diff
@@ -1,37 +1,75 @@
 import { Injectable } from '@angular/core';
 import { IBook } from '../models/book.model';
-import { ICart } from '../models/cart.model';
+import { ICart, ICartItem } from '../models/cart.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CartService {
-  booksInCart: ICart[] = [];
+  cart: ICart = {
+    cartItems: [],
+    totalQty: 0,
+    totalCost: 0,
+  };
 
-  getItems(): ICart[] {
-    return this.booksInCart;
+  getItems(): ICart {
+    return this.cart;
   }
 
-  addToCart(book: IBook): void {
-    const ind: number = this.booksInCart.findIndex((el) => el.book.id === book.id);
-    if (ind === -1) {
-      this.booksInCart.push({ book, qty: 1 });
+  getItemIndx(id: number): number {
+    const itemIndx = this.cart.cartItems.findIndex((cartItem) => {
```

Comment:
You can write it like this, it's a little bit shorter

`const itemIndx = this.cart.cartItems.findIndex(({ id: cartId }) => id === cartId);`

Reply:
Method deleted, as I don't use it anymore.